lib/Drush/Boot/DrupalBoot8.php

```diff
@@ -118,15 +120,26 @@ function bootstrap_drupal_database() {
   }
 
   function bootstrap_drupal_configuration() {
+    global $drupal_update;
     $this->request = Request::createFromGlobals();
     $classloader = drush_drupal_load_autoloader(DRUPAL_ROOT);
     // @todo - use Request::create() and then no need to set PHP superglobals
     $kernelClass = new \ReflectionClass('\Drupal\Core\DrupalKernel');
     if ($kernelClass->hasMethod('addServiceModifier')) {
-      $this->kernel = DrupalKernel::createFromRequest($this->request, $classloader, 'prod', DRUPAL_ROOT);
+      if (!$drupal_update) {
+        $this->kernel = DrupalKernel::createFromRequest($this->request, $classloader, 'prod');
+      }
+      else {
+        $this->kernel = UpdateKernel::createFromRequest($this->request, $classloader, 'prod');
+      }
     }
     else {
-      $this->kernel = DrushDrupalKernel::createFromRequest($this->request, $classloader, 'prod', DRUPAL_ROOT);
+      if (!$drupal_update) {
+        $this->kernel = DrushDrupalKernel::createFromRequest($this->request, $classloader, 'prod');
+      }
+      else {
+        $this->kernel = DrushDrupalUpdateKernel::createFromRequest($this->request, $classloader, 'prod');
```

Comment:
The UpdateKernel is part of Drupal since the 8.0.0 release, but the DrupalKernel and the UpdateKernel both don't have an addServiceModifier() method. I have no idea why drush is implementing an adapter to add this method and add a service modifier through it. The DrupalKernel way of adding service modifiers is through A) defining in a module ModuleNameServiceProvider class or B) putting the class or the service object under $GLOBALS['conf']['container_service_providers'].